plugins/chat/assets/javascripts/discourse/lib/staged-message-handler.js

```diff
@@ -0,0 +1,28 @@
+export function handleStagedMessage(messagesManager, data) {
```

Comment:
Could probably have this somewhere nicer, for now was easiest to just extract it out.